stubs/regex/regex/regex.pyi

```diff
@@ -0,0 +1,349 @@
+def sub(
+    pattern: AnyStr,
+    repl: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def sub(
+    pattern: _regex.Pattern,
+    repl: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subf(
+    pattern: AnyStr,
+    format: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subf(
+    pattern: _regex.Pattern,
+    format: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subn(
+    pattern: AnyStr,
+    repl: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subn(
+    pattern: _regex.Pattern,
+    repl: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> AnyStr: ...
+@overload
+def subfn(
+    pattern: AnyStr,
+    format: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> tuple[AnyStr, int]: ...
+@overload
+def subfn(
+    pattern: _regex.Pattern,
+    format: AnyStr | Callable[[_regex.Match], AnyStr],
+    string: AnyStr,
+    count: int = ...,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> tuple[AnyStr, int]: ...
+
+Pattern: _regex.Pattern
+Match: _regex.Match
```

Comment:
```suggestion
Pattern: type[_regex.Pattern]
Match: type[_regex.Match]
```

Reply:
Would a type alias work here?

Reply:
both methods seem to work thanks! I've gone with the type alias:
```python
Pattern = _regex.Pattern
Match = _regex.Match
```